### DISCUSSION TITLE
Problemas con Maven

### DISCUSSION THREAD

**Discussion by ezecastiglione18** [dds-jv/foro#194] [Consultas de herramientas]
Buenas! Quise correr mvn test con el projecto y me tira esto: 
![image](https://user-images.githubusercontent.com/26487111/115794698-d47f2b00-a3a4-11eb-8f32-514852c28883.png)

Alguno podria ayudarme con esto?

**Comment by gastonprieto** [comment]
Fijate que esta ejecutando la jdk 16, podras probar con la jdk8 para ver si con eso deja de fallar?

**Comment by ezecastiglione18** [comment]
Nop, no se soluciono. Es mas, creo que esta peor. Porque directamente ni el cmd ni git bash no me devuelven nada jajajaj 

![image](https://user-images.githubusercontent.com/26487111/115926917-00a8b380-a45a-11eb-948c-bbfbbd354fa6.png)

Y ya hice descargue el jdk 8: 
![image](https://user-images.githubusercontent.com/26487111/115926967-14ecb080-a45a-11eb-87d0-2df5e2cf56d8.png)

Y ya cree la variable de entorno: 

![image](https://user-images.githubusercontent.com/26487111/115927067-406f9b00-a45a-11eb-84f0-62e4e05e3222.png)

**Reply by gastonprieto** [reply]
Hola de nuevo

El comando para saber que java esta ejecutando es: `java --version`

Fijate que además de agregar la `JAVA_HOME` tenes que agregarlo a `PATH` que es donde busca todos los comandos para ejecutar.

**Reply by ezecastiglione18** [reply]
Perdona que responda recien ahora, pero eso del PATH no entendi muy bien que hacer.

Y con respecto a lo otro: ejecute "java --version" y me sigue pasando lo mismo. Como que la consola no sabe que responder

**Reply by nanderson87** [reply]
Buenas, para agregar java a `PATH` tenes que ir a la misma sección donde agregaste la variable `JAVA_HOME` de dentro _Variables del sistema_.

Si ves el listado de _Variables del sistema_ hay una que se llama justamente `PATH`
![image](https://user-images.githubusercontent.com/11875266/116003047-c7e01a00-a5d2-11eb-85dc-c9ee621b1185.png)

La seleccionas >> presionas el botón editar >> Eso te va a abrir una nueva ventana >> Le das al botón nuevo >> Agregas la ruta hasta la carpeta bin de tu JDK en tu caso `C:\Program Files\Java\jdk1.8.0_291\bin` >> Una vez agregado le das al botón OK/Aceptar. 


Una vez realizado eso debería ir de nuevo a la consola/terminal que estes usando y volver a probar el comando `java -version`.  Si te sigue sin mostrar nada o te tira algun error. Te pido que nos adjuntes una imagen de lo que tengas configurado en `PATH`

**Comment by ezecastiglione18** [comment]
Buenas! Sorry que responda ahora.

Ya probe con -version y --version y nada. Esto es lo que tengo en el Path
de la Variables de Sistema:

![image](https://user-images.githubusercontent.com/26487111/116013777-4e631e80-a608-11eb-94dd-a474f2e681bd.png)


Igual ya agregue un par de tests mas y funcionan todos. Lo subo igual?

**Reply by nanderson87** [reply]
> Igual ya agregue un par de tests mas y funcionan todos. Lo subo igual?

Respecto a esto si te referis a que salio todo bien ejecutando `mvn test`. Si, entonces podes continuar con el comando `mvn verify` (verifica algunas cosas más, como la cobertura de tu repo) y si eso sale bien ejecuta el comando para subir el repo.


> Ya probe con -version y --version y nada. Esto es lo que tengo en el Path
> de la Variables de Sistema:

Respecto a esto te invito a hacer dos pruebas más, en el orden que aparece a continuación:
1. Dentro del editor de variables de entorno. Con los botones subir y bajar, deja más arriba la entrada que tiene la carpeta bin de la jdk (la que creaste vos) respecto a las entradas que aputan a la carpeta _{algo}\Oracle\Java\javapath_ (hay dos).
2. Si eso no llega a funcionar directamente  elimina las entradas del estilo _Oracle\Java\javapath_ dejando sola la que apunta a la carpeta bin de la jdk. (haciendo un previo backup de esas entradas)

_Nota: Siempre que hagas un cambio en las variables de entorno y le des aceptar. Cerra y volve a abrir las consolas para que impacte el cambio._

Si ninguna de esas  dos cosas no funcionan. Dejalo todo como estaba en la imagen que nos pasaste.

**Reply by ezecastiglione18** [reply]
Ya probe esas dos cosas y nada. Asi que directamente subo el tp y nos olvidamos de esto. Gracias por la ayuda!
